src/index.tsx

```diff
@@ -436,43 +490,54 @@ export class ScrollBottomSheet<T extends any> extends Component<Props<T>> {
             this.calculateNextSnapPoint(i + 1)
           );
 
-    const runTiming = ({
+    const runAnimation = ({
       clock,
       from,
       to,
       position,
       finished,
+      velocity,
       frameTime,
     }: TimingParams) => {
       const state = {
         finished,
+        velocity: new Value(0),
         position,
         time: new Value(0),
         frameTime,
       };
 
-      const animationParams = {
+      const timingConfig = {
         duration: animationDuration,
-        easing: animationConfig?.easing || DEFAULT_EASING,
+        easing:
+          (props.animationType === 'timing' && props.animationConfig?.easing) ||
+          DEFAULT_EASING,
+        toValue: new Value(0),
       };
 
-      const config = {
+      const springConfig = {
+        ...((props.animationType === 'spring' && props.animationConfig) || DEFAULT_SPRING_PARAMS),
         toValue: new Value(0),
-        ...animationParams,
       };
```

Comment:
We need to merge the provided spring config with the defaults, just in case user only wants to customise some of the values.

```ts
      const springConfig = {
		...DEFAULT_SPRING_PARAMS,
        ...((props.animationType === 'spring' && props.animationConfig) ||
          DEFAULT_SPRING_PARAMS),
        toValue: new Value(0),
      };
```